### ISSUE TITLE
About acwgan-gp

### ISSUE THREAD

**Issue by upgf** [lenalee103/Aggregates-Database-based-on-ACGAN#2]
Thank you very much for open-sourcing your code！Have you tried your code on other data ? I have met some problems training my own data with it. In the first few hundred epochs, the true and false accuracy will fluctuate around 50%, but it will gradually decrease to 0.00%, and the network cannot converge to 50% accuracy again. Can you give me some advice?

**Comment by lenalee103** [comment]
Hello, have you check the loss, it can show some clues. My code is only used for the pavement aggregates, and it has a classification aside in the discriminator network. If you can Tests the discriminator and generator loss separately, it will tell where is the problem.---- Replied Message ***@***.***>Date03/07/2022 15:35 ***@***.***> ***@***.***>Subject[lenalee103/Aggregates-Database-based-on-ACGAN] About acwgan-gp (Issue #2)

Thank you very much for open-sourcing your code！Have you tried your code on other data ? I have met some problems training my own data with it. In the first few hundred epochs, the true and false accuracy will fluctuate around 50%, but it will gradually decrease to 0.00%, and the network cannot converge to 50% accuracy again. Can you give me some advice?



—Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[

{

***@***.***": "http://schema.org",

***@***.***": "EmailMessage",

"potentialAction": {

***@***.***": "ViewAction",

"target": "https://github.com/lenalee103/Aggregates-Database-based-on-ACGAN/issues/2",

"url": "https://github.com/lenalee103/Aggregates-Database-based-on-ACGAN/issues/2",

"name": "View Issue"

},

"description": "View this Issue on GitHub",

"publisher": {

***@***.***": "Organization",

"name": "GitHub",

"url": "https://github.com"

}

}

]

**Comment by upgf** [comment]
作业已收到，谢谢。

**Comment by upgf** [comment]
Thanks for your reply! I observed my d_loss and g_loss values and found that d_loss is fluctuating and cannot converge as well as g_loss. After going through some posts, I tried to change BatchNormalization to LayerNormalization, but failed due to&nbsp;the version issue of Keras, and removing sigmod didn't help my problem either.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "lenalee103/Aggregates-Database-based-on-ACGAN"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年3月7日(星期一) 晚上10:58
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [lenalee103/Aggregates-Database-based-on-ACGAN] About acwgan-gp (Issue #2)





 
             Hello, have you check the loss, it can show some clues. My code is only used for the pavement aggregates, and it has a classification aside in the discriminator network. If you can Tests the discriminator and generator loss separately, it will tell where is the problem.---- Replied Message ***@***.***&gt;Date03/07/2022 15:35 ***@***.***&gt; ***@***.***&gt;Subject[lenalee103/Aggregates-Database-based-on-ACGAN] About acwgan-gp (Issue #2)
 
 Thank you very much for open-sourcing your code！Have you tried your code on other data ? I have met some problems training my own data with it. In the first few hundred epochs, the true and false accuracy will fluctuate around 50%, but it will gradually decrease to 0.00%, and the network cannot converge to 50% accuracy again. Can you give me some advice?
 
 
 
 —Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.
 
 You are receiving this because you are subscribed to this thread.Message ID: ***@***.***&gt;
 
 [
 
 {
 
 ***@***.***": "http://schema.org",
 
 ***@***.***": "EmailMessage",
 
 "potentialAction": {
 
 ***@***.***": "ViewAction",
 
 "target": "https://github.com/lenalee103/Aggregates-Database-based-on-ACGAN/issues/2",
 
 "url": "https://github.com/lenalee103/Aggregates-Database-based-on-ACGAN/issues/2",
 
 "name": "View Issue"
 
 },
 
 "description": "View this Issue on GitHub",
 
 "publisher": {
 
 ***@***.***": "Organization",
 
 "name": "GitHub",
 
 "url": "https://github.com"
 
 }
 
 }
 
 ] 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by lenalee103** [comment]
过拟合了…本来 d和g就会有不平衡问题…需要简化d，我是用了wasserstein loss，减少了不平衡问题…你可以搜搜看不平衡问题还有什么好的解决方案---- Replied Message ***@***.***>Date03/08/2022 03:04 ***@***.***> ***@***.******@***.***>SubjectRe: [lenalee103/Aggregates-Database-based-on-ACGAN] About acwgan-gp (Issue #2)

Thanks for your reply! I observed my d_loss and g_loss values and found that d_loss is fluctuating and cannot converge as well as g_loss. After going through some posts, I tried to change BatchNormalization to LayerNormalization, but failed due to&nbsp;the version issue of Keras, and removing sigmod didn't help my problem either.



















------------------&nbsp;原始邮件&nbsp;------------------



发件人:                                                                                                                        "lenalee103/Aggregates-Database-based-on-ACGAN"                                                                                    ***@***.***&gt;;



发送时间:&nbsp;2022年3月7日(星期一) 晚上10:58



***@***.***&gt;;



***@***.******@***.***&gt;;



主题:&nbsp;Re: [lenalee103/Aggregates-Database-based-on-ACGAN] About acwgan-gp (Issue #2)























  



             Hello, have you check the loss, it can show some clues. My code is only used for the pavement aggregates, and it has a classification aside in the discriminator network. If you can Tests the discriminator and generator loss separately, it will tell where is the problem.---- Replied Message ***@***.***&gt;Date03/07/2022 15:35 ***@***.***&gt; ***@***.***&gt;Subject[lenalee103/Aggregates-Database-based-on-ACGAN] About acwgan-gp (Issue #2)



  



 Thank you very much for open-sourcing your code！Have you tried your code on other data ? I have met some problems training my own data with it. In the first few hundred epochs, the true and false accuracy will fluctuate around 50%, but it will gradually decrease to 0.00%, and the network cannot converge to 50% accuracy again. Can you give me some advice?



  



  



  



 —Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.



  



 You are receiving this because you are subscribed to this thread.Message ID: ***@***.***&gt;



  



 [



  



 {



  



 ***@***.***": "http://schema.org",



  



 ***@***.***": "EmailMessage",



  



 "potentialAction": {



  



 ***@***.***": "ViewAction",



  



 "target": "https://github.com/lenalee103/Aggregates-Database-based-on-ACGAN/issues/2",



  



 "url": "https://github.com/lenalee103/Aggregates-Database-based-on-ACGAN/issues/2",



  



 "name": "View Issue"



  



 },



  



 "description": "View this Issue on GitHub",



  



 "publisher": {



  



 ***@***.***": "Organization",



  



 "name": "GitHub",



  



 "url": "https://github.com"



  



 }



  



 }



  



 ]  



—



Reply to this email directly, view it on GitHub, or unsubscribe.



Triage notifications on the go with GitHub Mobile for iOS or Android.  



You are receiving this because you authored the thread.Message ID: ***@***.***&gt;



—Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

You are receiving this because you commented.Message ID: ***@***.***>

[

{

***@***.***": "http://schema.org",

***@***.***": "EmailMessage",

"potentialAction": {

***@***.***": "ViewAction",

"target": "https://github.com/lenalee103/Aggregates-Database-based-on-ACGAN/issues/2#issuecomment-1061330524",

"url": "https://github.com/lenalee103/Aggregates-Database-based-on-ACGAN/issues/2#issuecomment-1061330524",

"name": "View Issue"

},

"description": "View this Issue on GitHub",

"publisher": {

***@***.***": "Organization",

"name": "GitHub",

"url": "https://github.com"

}

}

]

**Comment by upgf** [comment]
OK, I&nbsp;will try it. Thank you very much for your reply.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "lenalee103/Aggregates-Database-based-on-ACGAN"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年3月8日(星期二) 中午1:40
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [lenalee103/Aggregates-Database-based-on-ACGAN] About acwgan-gp (Issue #2)





 过拟合了…本来 d和g就会有不平衡问题…需要简化d，我是用了wasserstein loss，减少了不平衡问题…你可以搜搜看不平衡问题还有什么好的解决方案---- Replied Message ***@***.***&gt;Date03/08/2022 03:04 ***@***.***&gt; ***@***.******@***.***&gt;SubjectRe: [lenalee103/Aggregates-Database-based-on-ACGAN] About acwgan-gp (Issue #2)
 
 Thanks for your reply! I observed my d_loss and g_loss values and found that d_loss is fluctuating and cannot converge as well as g_loss. After going through some posts, I tried to change BatchNormalization to LayerNormalization, but failed due to&amp;nbsp;the version issue of Keras, and removing sigmod didn't help my problem either.
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 ------------------&amp;nbsp;原始邮件&amp;nbsp;------------------
 
 
 
 发件人:                                                                                                                        "lenalee103/Aggregates-Database-based-on-ACGAN"                                                                                    ***@***.***&amp;gt;;
 
 
 
 发送时间:&amp;nbsp;2022年3月7日(星期一) 晚上10:58
 
 
 
 ***@***.***&amp;gt;;
 
 
 
 ***@***.******@***.***&amp;gt;;
 
 
 
 主题:&amp;nbsp;Re: [lenalee103/Aggregates-Database-based-on-ACGAN] About acwgan-gp (Issue #2)
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
   
 
 
 
              Hello, have you check the loss, it can show some clues. My code is only used for the pavement aggregates, and it has a classification aside in the discriminator network. If you can Tests the discriminator and generator loss separately, it will tell where is the problem.---- Replied Message ***@***.***&amp;gt;Date03/07/2022 15:35 ***@***.***&amp;gt; ***@***.***&amp;gt;Subject[lenalee103/Aggregates-Database-based-on-ACGAN] About acwgan-gp (Issue #2)
 
 
 
   
 
 
 
  Thank you very much for open-sourcing your code！Have you tried your code on other data ? I have met some problems training my own data with it. In the first few hundred epochs, the true and false accuracy will fluctuate around 50%, but it will gradually decrease to 0.00%, and the network cannot converge to 50% accuracy again. Can you give me some advice?
 
 
 
   
 
 
 
   
 
 
 
   
 
 
 
  —Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.
 
 
 
   
 
 
 
  You are receiving this because you are subscribed to this thread.Message ID: ***@***.***&amp;gt;
 
 
 
   
 
 
 
  [
 
 
 
   
 
 
 
  {
 
 
 
   
 
 
 
  ***@***.***": "http://schema.org",
 
 
 
   
 
 
 
  ***@***.***": "EmailMessage",
 
 
 
   
 
 
 
  "potentialAction": {
 
 
 
   
 
 
 
  ***@***.***": "ViewAction",
 
 
 
   
 
 
 
  "target": "https://github.com/lenalee103/Aggregates-Database-based-on-ACGAN/issues/2",
 
 
 
   
 
 
 
  "url": "https://github.com/lenalee103/Aggregates-Database-based-on-ACGAN/issues/2",
 
 
 
   
 
 
 
  "name": "View Issue"
 
 
 
   
 
 
 
  },
 
 
 
   
 
 
 
  "description": "View this Issue on GitHub",
 
 
 
   
 
 
 
  "publisher": {
 
 
 
   
 
 
 
  ***@***.***": "Organization",
 
 
 
   
 
 
 
  "name": "GitHub",
 
 
 
   
 
 
 
  "url": "https://github.com"
 
 
 
   
 
 
 
  }
 
 
 
   
 
 
 
  }
 
 
 
   
 
 
 
  ]  
 
 
 
 —
 
 
 
 Reply to this email directly, view it on GitHub, or unsubscribe.
 
 
 
 Triage notifications on the go with GitHub Mobile for iOS or Android.  
 
 
 
 You are receiving this because you authored the thread.Message ID: ***@***.***&amp;gt;
 
 
 
 —Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.
 
 You are receiving this because you commented.Message ID: ***@***.***&gt;
 
 [
 
 {
 
 ***@***.***": "http://schema.org",
 
 ***@***.***": "EmailMessage",
 
 "potentialAction": {
 
 ***@***.***": "ViewAction",
 
 "target": "https://github.com/lenalee103/Aggregates-Database-based-on-ACGAN/issues/2#issuecomment-1061330524",
 
 "url": "https://github.com/lenalee103/Aggregates-Database-based-on-ACGAN/issues/2#issuecomment-1061330524",
 
 "name": "View Issue"
 
 },
 
 "description": "View this Issue on GitHub",
 
 "publisher": {
 
 ***@***.***": "Organization",
 
 "name": "GitHub",
 
 "url": "https://github.com"
 
 }
 
 }
 
 ] 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
